### DISCUSSION TITLE
Record Attribute for ignoring equality checks.

### DISCUSSION THREAD

**Discussion by MichaelPuckett2** [dotnet/csharplang#7114] [Language Ideas]
Records are my favorite type now for immutability and value comparisons but I've ran into a small issue.

If I make a record that's not immutable because I want to use the value comparison easily and if that record has an event (for example) then the value comparison fails when someone subscribes to the event in one of the records. 

I also, almost always add a static Empty property to records to have a base for comparison to remove nulls and give me a baseline to compare to.  It would be cool if something like this could also be added to records but for the sake of this question continued below.

I was wondering if we could add an attribute to a property (or event in this example) that would exclude it from comparisons.
_My example shows the dilemma, not the idea for a fix._

Example:
```
var person1 = new Person("", "");
var person2 = person1 with { };
var person3 = Person.Empty;

Console.WriteLine(person1 == person2);
Console.WriteLine(person1 == person3);
Console.WriteLine(person2 == person3);
//Output:
//true
//true
//true

person1.Updated += () => { };

Console.WriteLine(person1 == person2);
Console.WriteLine(person1 == person3);
Console.WriteLine(person2 == person3);
//Output:
//false
//false
//true

public record Person(string FirstName, string LastName)
{
    public event Action Updated = delegate { };
    public static Person Empty { get; } = new(string.Empty, string.Empty);
}
```

**Comment by CyrusNajmabadi** [comment]
> Record Attribute for ignoring equality checks.

You can def add an attribute for this.  :)   You'll just need to define such an attribute, and then write a Source-Generator to generate *different* equality methods into another 'partial part' of your record.  These equality methods will then cause the default ones to not be created.

For writing a source-generator, use the [ForAttributeWithMetadataName](https://github.com/dotnet/roslyn/blob/main/src/Compilers/Core/Portable/SourceGeneration/Nodes/SyntaxValueProvider_ForAttributeWithMetadataName.cs#L78) so that you are only called on your records that you want to override the semantics for.  

--

Side note: i would recommend not having events in your records.  It's def wonky for things that are intended to represent pure-values.  Instead, have the core record without events, and then create a composite type that has the event and poitns at the record (but which does not pretend to have value semantics itself).
